Framework/PythonInterface/plugins/algorithms/CropWorkspaceRagged.py

```diff
@@ -0,0 +1,122 @@
+#pylint: disable=no-init,invalid-name
+from __future__ import (absolute_import, division, print_function)
+from mantid.api import *
+from mantid.simpleapi import ConjoinWorkspaces, CropWorkspace, RenameWorkspace
+from mantid.kernel import *
+import numpy as np
+
+
+class CropWorkspaceRagged(PythonAlgorithm):
+
+    def category(self):
+        return 'Transforms\\Splitting;Workflow'
+
+    def name(self):
+        return 'CropWorkspaceRagged'
+
+    def summary(self):
+        return 'Crop each spectrum of a workspace independently'
+
+    def PyInit(self):
+        self.declareProperty(WorkspaceProperty('InputWorkspace', '',
+                                               direction=Direction.Input), 'input workspace')
+        self.declareProperty(WorkspaceProperty('OutputWorkspace', '',
+                                               direction=Direction.Output), 'output workspace')
+        self.declareProperty(FloatArrayProperty('Xmin'), 'minimum x values with NaN meaning no minimum')
+        self.declareProperty(FloatArrayProperty('Xmax'), 'maximum x values with NaN meaning no maximum')
```

Comment:
Is it necessary to accept NaNs here? Wouldn't `-np.inf` and `np.inf` do the job as well and be more logical?